src/Core/src/Handlers/Button/ButtonHandler.iOS.cs

```diff
@@ -202,9 +211,56 @@ public override void SetImageSource(UIImage? platformImage)
 				if (Handler?.PlatformView is not UIButton button)
 					return;
 
-				platformImage = platformImage?.ImageWithRenderingMode(UIImageRenderingMode.AlwaysOriginal);
+				nfloat buttonSize = 0;
+
+#pragma warning disable CA1422 // Validate platform compatibility
+				var contentEdgeInsets = button.ContentEdgeInsets;
+#pragma warning restore CA1422 // Validate platform compatibility
+
+				if (button.Bounds != CGRect.Empty)
+				{
+					if (button.Bounds.Height > button.Bounds.Width)
+						buttonSize = button.Bounds.Width - (contentEdgeInsets.Left + contentEdgeInsets.Right);
+					else
+						buttonSize = button.Bounds.Height - (contentEdgeInsets.Top + contentEdgeInsets.Bottom);
+				}
+
+				try
+				{
+					if (buttonSize != 0)
+						platformImage = ResizeImageSource(platformImage, buttonSize, buttonSize);
+
+					platformImage = platformImage?.ImageWithRenderingMode(UIImageRenderingMode.AlwaysOriginal);
+
+					button.SetImage(platformImage, UIControlState.Normal);
+				}
+				catch (Exception)
+				{
+					Handler.MauiContext?.CreateLogger<ButtonHandler>()?.LogWarning("Can not load Button ImageSource");
+				}
+			}
+
+			static UIImage? ResizeImageSource(UIImage? sourceImage, nfloat maxWidth, nfloat maxHeight)
+			{
+				if (sourceImage is null)
+					return null;
+
+				var sourceSize = sourceImage.Size;
+				var maxResizeFactor = Math.Min(maxWidth / sourceSize.Width, maxHeight / sourceSize.Height);
+
+				if (maxResizeFactor > 1)
+					return sourceImage;
+
+				var width = maxResizeFactor * sourceSize.Width;
+				var height = maxResizeFactor * sourceSize.Height;
+				CGSize scaleSize = new CGSize(width, height);
+
+				UIGraphics.BeginImageContextWithOptions(new CGSize((nfloat)width, (nfloat)height), false, 0);
+				sourceImage.Draw(new CGRect(0, 0, scaleSize.Width, scaleSize.Height));
+				var resultImage = UIGraphics.GetImageFromCurrentImageContext();
+				UIGraphics.EndImageContext();
```

Comment:
These re-drawings will break the gif support as well: https://github.com/dotnet/maui/pull/19411

Reply:
In some of my testing just now, I see we can not actually resize the data, but lie to UIKit about the real size:

```cs
resultImage = UIImage.FromImage(sourceImage.CGImage, sourceImage.Scale / maxResizeFactor, maxResizeFactor.Orientation); 
```

Reply:
The scale parameter is the inverse of normal scale as it is the density. So a 100x100 pixel image with a scale of 1 is 100x100 and a scale of 2 is actually going to be drawn at 50x50.

Reply:
Applied changes to avoid recreate  the image.